File: python/7 kyu/374.py
# https://www.codewars.com/kata/564e22e30e686c7f7e00005d/train/python

# Passed

def knight_vs_rook(knight, rook):
    knight_square, knight_file = knight
    rook_square, rook_file = rook
    
    knight_moves = []
    for direction in [1, -1]:
        knight_moves.extend(
            [   
                (knight_square + direction, chr(ord(knight_file) + 2)),
                (knight_square + direction, chr(ord(knight_file) - 2)),
                (knight_square + 2 * direction, chr(ord(knight_file) + 1)),
                (knight_square + 2 * direction, chr(ord(knight_file) - 1))
            ]
        )
    if knight_file == rook_file or knight_square == rook_square:
        result = "Rook"
    elif any(knight_move == rook for knight_move in knight_moves):
        result = "Knight"
    else:
        result = "None"
        
    return result

output = knight_vs_rook((6,'D'), (4,'C'))
print(output)

File: nodejs/7 kyu/374.js
// https://www.codewars.com/kata/5a0efbb7c374cb69970000cf/train/javascript

// Passed

function reverseMessage(str) {
  let result = "";
  if (str.length !== 0) {
    let words = str.split(" ");
    let toTitleCase = (word) =>
      word[0].toUpperCase() + word.slice(1).toLowerCase();
    let wordsToJoin = words
      .reverse()
      .map((word) => toTitleCase(word.split("").reverse().join("")));
    result = wordsToJoin.join(" ");
  }
  return result;
}

const output = reverseMessage("Reverse this message!");
console.log(output);